### PR TITLE
Add skill to convert date to big/little-endian format

### PR DESCRIPTION
**Pull request by krcmarik**
**Describe the contribution to the taxonomy**

<!-- A concise description of what the contribution brings, replace "..." in the bullet list -->

- Convert a given date to the big-endian date format
- Or convert a given date to the little-endian date format

**Input given at the prompt**

<!-- What you entered, replace "..." -->

```
>>> Convert little-endian date 1st June 2022 to big-endian format.                                                                                                                             
  
>>> Convert the given date, June 1st 2023, to the big-endian date format                                 

>>> Convert the given date, 16th December 1999, to the little-endian date format

>>> Convert the given date, 2021-05-26, to the little-endian date format.                                                                                                                                                                                                                                              

>>> Given the date December 31st, 2010, determine the year in the big-endian format.                                                                                                                                             

```


**Response from the original model**


<!-- What you received from the original model in response to your input, 
replace "..." -->

```
1. To help you convert a date from little-endian format to big-endian format, I'd need to know the specific date format you're using. However, based on your latest input, I can assume that 1st June 2022 represents the date June 1st, 2022 in little-endian format (day, month, year). To convert this date to big-endian format (year, month, day), it would be June 1st, 2022. If you'd like to convert another date or need assistance with date conversions in the future, feel free to ask!

2. To convert the given date, June 1st 2023, to the big-endian date format, we first need to identify the day, month, and year components of the given date. In this case, the day is 1st, the month is June, and the year is 2023. To convert the given date to the big-endian date format, we simply reorder the day, month, and year components as day, month, year. So, June 1st 2023 in the big-endian date format would be represented as: 010620, which means that the 1st of June 2023 is the 010620th day of the year.

3. To convert the given date, 16th December 1999, to the little-endian date format, we first need to identify the day, month, and year components of the given date. In this case, the day is 16th, the month is December, and the year is 1999. To convert the given date to the little-endian date format, we simply reorder the day, month, and year components as year, month, day. So, the 16th of December 1999 in the little-endian date format would be represented as: 991216, which means that the 16th of December 1999 is the 991216th day of the year.

4. In the little-endian date format, the year is represented by the last two digits, and the month and day are swapped compared to the standard format. To convert the given date, 2021-05-26, into the little-endian format, follow these steps:                                                                                                                                             
  1. Extract the last two digits of the year (i.e., 21 from 2021).
  2. Swap the month and day (i.e., replace 05 with 26 and leave the year as 21).
So, the little-endian format for the date 2021-05-26 is 21-26 (year-day).

5. To determine the year in the big-endian format, we simply concatenate the year, month, and day in that order. The year is represented in the range of 70 to 99 (that is, 70 to 99). So, let's adjust the input year accordingly: Input year: 2010 - 1970 = 40
```


**Response from the fine-tuned model**


<!-- Generate a synthetic dataset based on your newly added seed data; train the model 
with the synthetic data and now re-test the model's response with the same prompt.
Replace "..." with what you receive with the finetuned model. -->

```
 No suitable hardware available
```

**Contribution checklist**

<!-- Insert an x between the empty brackets: [ ] >> [x] -->

- [x] The contribution was tested with `lab generate`
- [x] No errors or warnings were produced by `lab generate`
- [x] All [commits are signed off](https://github.com/instruct-lab/taxonomy/blob/main/CONTRIBUTING.md#legal) (DCO)
- [x] The `qna.yaml` file contains at least 5 `seed_examples`
- [x] The `qna.yaml` file was [linted](https://yamllint.com) and [prettified](https://onlineyamltools.com/prettify-yaml) ([yaml-validator](https://jsonformatter.org/yaml-validator) can do both)

### CODE REVIEW

**Review by mingxzhao — COMMENTED** [review]
Looks great! I will mark as approved. Do you think it would be useful to also include some seed examples that use the numerical date format as well? This may improve the responses by adding some more diversity.

**Comment by n1hility** [comment]
Thank you for your contribution to InstructLab! Unfortunately, once a Github repo is made public, all open PRs are automatically closed since they link against a private repo. We have detected that your PR might have been one of the ones affected by this change. If you are still interested in contributing your improvement, please fill out the following short form by no later than May 3rd, and we will get back to you with the additional steps necessary once we have had time to assess the PRs of those still interested:

https://forms.gle/V7SrPPMZDo6iGDYu8
